src/testing/advection_explicit_periodic.cpp

```diff
@@ -10,180 +10,360 @@
 #include <deal.II/grid/grid_tools.h>
 #include <deal.II/grid/grid_out.h>
 #include <deal.II/grid/grid_in.h>
+#include <deal.II/base/convergence_table.h>
 
 #include "parameters/all_parameters.h"
 #include "parameters/parameters.h"
 #include "physics/physics_factory.h"
 #include "physics/physics.h"
+#include "dg/dg.h"
 #include "dg/dg_factory.hpp"
 #include "ode_solver/ode_solver_factory.h"
 #include "advection_explicit_periodic.h"
+#include "physics/initial_conditions/initial_condition_base.h"
+
+#include "mesh/grids/nonsymmetric_curved_periodic_grid.hpp"
 
 #include<fenv.h>
 
+#include <deal.II/grid/manifold_lib.h>
+#include <deal.II/fe/mapping_q.h>
+#include <fstream>
+
 namespace PHiLiP {
 namespace Tests {
 template <int dim, int nstate>
 AdvectionPeriodic<dim, nstate>::AdvectionPeriodic(const PHiLiP::Parameters::AllParameters *const parameters_input)
 :
 TestsBase::TestsBase(parameters_input)
+, mpi_communicator(MPI_COMM_WORLD)
+, pcout(std::cout, dealii::Utilities::MPI::this_mpi_process(mpi_communicator)==0)
 {}
 
+template<int dim, int nstate>
+double AdvectionPeriodic<dim, nstate>::compute_energy(std::shared_ptr < PHiLiP::DGBase<dim, double> > &dg) const
```

Comment:
We should discuss with Doug about passing the shared point of the object by reference.

Reply:
https://stackoverflow.com/questions/47609291/c-is-it-ok-to-use-shared-pointers-in-a-function-as-parameters-or-am-i-wasting

Reply:
Passing shared_ptr by reference makes no sense. It's like passing a function(const &* int input). It would be like a reference to the pointer, which points to the object you are interested in.

Smart pointers like shared_ptr and unique_ptr only make sense when you are storing it somewhere. If you pass it to a function, pass it by const reference `compute_energy(PHiLiP::DGBase<dim, double> &dg)`.

Reply:
That link clarifies it quite well. @dougshidong , do you have any suggestions which syntax will be better to use in future if we are just passing the pointer but not storing it as a class member?